src/skin/skinloader.cpp

```diff
@@ -173,9 +173,122 @@ QWidget* SkinLoader::loadConfiguredSkin(QWidget* pParent,
+    // This is 1 if [Skin], show_spinnies == 1 OR [Skin],show_coverart == 1
+    // E.g. used to show/hide the skin settings toggle "Big spinny/cover"
+    ControlObject* pShowSpinnyAndOrCover = ControlObject::getControl(
+            ConfigKey("[Skin]", "show_spinny_or_cover"), ControlFlag::NoWarnIfMissing);
+    ControlObject* pSelectBigSpinnyCover = ControlObject::getControl(
+            ConfigKey("[Skin]", "select_big_spinny_or_cover"), ControlFlag::NoWarnIfMissing);
+    // This is set 1 if
+    //      [LateNight],show_spinny_cover == 1 AND [Skin],select_big_spinny_coverart == 0
+    ControlObject* pShowSmallSpinnyCover = ControlObject::getControl(
+            ConfigKey("[Skin]", "show_small_spinny_or_cover"), ControlFlag::NoWarnIfMissing);
+    // - [LateNight],show_big_spinny_cover: set 1 if
+    //      [LateNight],show_spinny_cover == 1 AND [Skin],select_big_spinny_coverart == 1
+    ControlObject* pShowBigSpinnyCover = ControlObject::getControl(
+            ConfigKey("[Skin]", "show_big_spinny_or_cover"), ControlFlag::NoWarnIfMissing);
+    // Check if the controls exist, i.e. were created by a skin (current or previous skins)
+    if (pShowSpinny &&
+            pShowCover &&
+            pShowSpinnyAndOrCover &&
+            pSelectBigSpinnyCover &&
+            pShowSmallSpinnyCover &&
+            pShowBigSpinnyCover) {
+        const QSet<ControlObject*> spinnyCoverTriggerControls = {
+                pShowSpinny,
+                pShowCover,
+                // pShowSpinnyAndOrCover,
+                pSelectBigSpinnyCover,
+                // pShowSmallSpinnyCover,
+                // pShowBigSpinnyCover
+                //  TODO(ronso0) Also connect controls consiered 'read-only'
+        };
+
+        std::function<void()> triggerFunc = [pShowSpinny,
+                                                    pShowCover,
+                                                    pShowSpinnyAndOrCover,
+                                                    pSelectBigSpinnyCover,
+                                                    pShowSmallSpinnyCover,
+                                                    pShowBigSpinnyCover]() {
+            pShowSpinnyAndOrCover->set(
+                    (pShowSpinny->get() > 0 ||
+                            pShowCover->get() > 0)
+                            ? 1.0
+                            : 0.0);
+            pShowSmallSpinnyCover->set(
+                    (pShowSpinnyAndOrCover->get() > 0 &&
+                            pSelectBigSpinnyCover->get() == 0)
+                            ? 1.0
+                            : 0.0);
+            pShowBigSpinnyCover->set(
+                    (pShowSpinnyAndOrCover->get() > 0 &&
+                            pSelectBigSpinnyCover->get() > 0)
+                            ? 1.0
+                            : 0.0);
+        };
+        triggerFunc();
+
+        for (auto* pCon : spinnyCoverTriggerControls) {
+            pCon->disconnect();
```

Comment:
This disconnects everything connected to the original control we are not owning. 
I think you can pass `Qt::UniqueConnection` below to get rid of this call.

Reply:
:+1: